src/features/defi/providers/idle/components/IdleManager/Claim/components/Confirm.tsx

```diff
@@ -0,0 +1,202 @@
+import { DefiParams, DefiQueryParams, DefiStep, DefiAction } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+
+export const Confirm = ({ onNext }: StepComponentProps) => {
+  const translate = useTranslate()
+  const { state, dispatch } = useContext(ClaimContext)
+  const { idle: idleInvestor } = useIdle()
+  const opportunity = state?.opportunity
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress: vaultAddress, assetReference } = query
+
+  const assetNamespace = 'erc20'
+
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference: vaultAddress,
+  })
+  
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace: 'slip44',
+    assetReference: ASSET_REFERENCE.Ethereum,
+  })
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  // user info
+  const { state: walletState } = useWallet()
+
+  const feeAssetBalance = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByAssetId(state, { assetId: feeAsset?.assetId ?? '' }),
+  )
+
+  useEffect(() => {
+    ;(async () => {
+      if (!dispatch || !state.userAddress || !idleInvestor) {
+        return
+      }
+      
+      dispatch({ type: IdleClaimActionType.SET_LOADING, payload: true })
+
+      const idleOpportunity = await idleInvestor.findByOpportunityId(assetId)
+      if (!idleOpportunity) throw new Error('No opportunity')
+
+      const preparedTx = await idleOpportunity.prepareClaimTokens(state.userAddress)
+      const estimatedGasCrypto = bnOrZero(preparedTx.gasPrice).times(preparedTx.estimatedGas).integerValue().toString()
+
+      dispatch({ type: IdleClaimActionType.SET_LOADING, payload: false })
+      dispatch({ type: IdleClaimActionType.SET_CLAIM, payload: { estimatedGasCrypto }})
+    })()
+  }, [state.userAddress, dispatch, idleInvestor, assetId])
+
+  let renderAssets: any[] = []
+  let claimableTokensTotalBalance = bnOrZero(0);
+
+  useAppSelector( selectorState => {
+    if (state && state.claimableTokens){
+      state.claimableTokens.forEach( token => {
+        const asset = selectAssetById(selectorState, token.assetId)
+        if (asset) {
+          claimableTokensTotalBalance = claimableTokensTotalBalance.plus(token.amount)
+          renderAssets.push((
+            <Stack direction='row' alignItems='center' justifyContent='center' key={token.assetId}>
+              <AssetIcon boxSize='8' src={asset.icon} />
+              <Amount.Crypto
+                fontSize='lg'
+                fontWeight='medium'
+                value={bnOrZero(token.amount).div(`1e+${asset.precision}`).toString()}
+                symbol={asset?.symbol}
+              />
+            </Stack>
+          ))
+        }
+      })
+    }
+  })
+
+  if (!state || !dispatch) return null
+
+  const handleConfirm = async () => {
```

Comment:
Since this is copied from the current implementation (which didn't wrap event handlers in callbacks) I'll take care of wrapping these in callbacks similar to https://github.com/shapeshift/web/pull/2325 after this is merged, let's not tackle it here as current implementation works (cc @0xdef1cafe)